src/plugins/vis_type_table/public/components/table_vis_component.tsx

```diff
@@ -31,23 +30,20 @@ export const TableVisComponent = ({
   event,
   uiState,
 }: TableVisComponentProps) => {
-  const { formattedRows: rows, formattedColumns: columns } = convertToFormattedData(
-    table,
-    visConfig
-  );
+  const { rows, columns } = table;
 
   const pagination = usePagination(visConfig, rows.length);
 
   const sortedRows = useMemo(() => {
-    return uiState.sort.colIndex !== null &&
+    return uiState.sort.colIndex !== undefined &&
       columns[uiState.sort.colIndex].id &&
       uiState.sort.direction
```

Comment:
```suggestion
    return columns[uiState.sort.colIndex]?.id !== undefined &&
      uiState.sort.direction
```
... since checking `uiState.sort.colIndex` proves nothing and all we care for is `columns[uiState.sort.colIndex].id` to exist.

Reply:
uiState.sort.colIndex is optional... shouldn't we request it to be not undefined? 
if not check this, there is a type error